ui/StatusQ/src/StatusQ/Components/StatusStepper.qml

```diff
@@ -0,0 +1,87 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+/*!
+   \qmltype StatusStepper
+   \inherits Item
+   \inqmlmodule StatusQ.Components
+   \since StatusQ.Components 0.1
+   \brief Displays total number of steps which need to be passed, marking each completed step
+          based on `completedSteps` property
+
+   Example:
+
+   \qml
+    StatusStepper {
+        width: 400
+        title: "Account %1 of %2"
+        totalSteps: 6
+        completedSteps: 1
+    }
+   \endqml
+
+   \image status_stepper.png
+
+   For a list of components available see StatusQ.
+*/
+
+Item {
+    id: root
+
+    /// Expected formatted text with %1 place marker for completed steps and %2 place marker for total steps count.
+    property string title: ""
+    property int titleFontSize: 12
+    property color titleColor: Theme.palette.baseColor1
+    property int totalSteps: 1
+    property int completedSteps: 1
+    property color completedStepColor: Theme.palette.primaryColor1
+    property color uncompletedStepColor: Theme.palette.baseColor2
+    property int leftPadding: 24
+    property int rightPadding: 24
+
+    implicitHeight: 52
+
+    QtObject {
+        id: d
+
+        readonly property int stepHeight: 4
+        readonly property int stepRadius: 4
+        readonly property int spaceBetweenSteps: 2
+    }
+
+    Column {
+        anchors.fill: parent
+        anchors.leftMargin: root.leftPadding
+        anchors.rightMargin: root.rightPadding
+        spacing: 8
+
+        StatusBaseText {
+            width: parent.width
+            horizontalAlignment: Qt.AlignHCenter
+            color: root.titleColor
+            font.pixelSize: root.titleFontSize
+            text: root.title.arg(root.completedSteps).arg(root.totalSteps)
```

Comment:
Then better expose this `text` property as a toplevel alias and assign where it's used from